src/entities/priceImpact/index.ts

```diff
@@ -200,6 +218,142 @@ export class PriceImpact {
         }
     };
 
+    /**
+     * Alternative for calculating price impact on add liquidity unbalanced operations
+     *
+     * Note: works on balancer v2 only because it relies on Remove Liquidity Unbalanced
+     * @param input same input used in the corresponding add liquidity operation
+     * @param poolState same pool state used in the corresponding add liquidity operation
+     * @returns price impact amount
+     */
+    static addLiquidityUnbalancedAlternative = async (
+        input: AddLiquidityUnbalancedInput,
+        poolState: PoolState,
+    ): Promise<PriceImpactAmount> => {
+        // inputs are being validated within AddLiquidity
+        if (poolState.balancerVersion !== 2) {
+            throw new Error(
+                'This alternative method relies on Remove Liquidity Unbalanced, which is only available for balancer V2.',
+            );
+        }
+
+        // simulate adding liquidity to get amounts in and bptOut
+        const addLiquidity = new AddLiquidity();
+        const { amountsIn, bptOut } = await addLiquidity.query(
+            input,
+            poolState,
+        );
+
+        // simulate removing liquidity exact out to get bptIn
+        const removeLiquidity = new RemoveLiquidity();
+        const removeLiquidityInput: RemoveLiquidityInput = {
+            chainId: input.chainId,
+            rpcUrl: input.rpcUrl,
+            amountsOut: amountsIn.map((a) => a.toInputAmount()),
+            kind: RemoveLiquidityKind.Unbalanced,
+        };
+        const { bptIn } = await removeLiquidity.query(
+            removeLiquidityInput,
+            poolState,
+        );
+
+        // get relevant amounts for price impact calculation
+        const amountInitial = parseFloat(formatUnits(bptOut.amount, 18));
```

Comment:
Is there a reason to use numbers here but bigint in `singleSwap`? Could maybe make a single helper ABA method on `PriceImpact` class that all could call and use consistent type?

Reply:
Yeah, you're right. A helper method does make sense for this. I'll do it over here.
For context, I wasn't worrying too much about numbers vs bigint precision because price impact is meant to be an approximation. But I agree that we should be consistent anyway. 👍